articles/20240820_gpu_utilization.md

```diff
@@ -0,0 +1,393 @@
+
+If [Docker](/definitions/20240819_definition_docker.md) isn't already installed on your system, follow these steps:
+
+```bash
+# Update package lists and install Docker
+sudo apt-get update
+sudo apt-get install docker.io
+sudo systemctl start docker
+```
+
+### NVIDIA Container Toolkit Installation
+
+The NVIDIA Container Toolkit is essential for enabling GPU support in [Docker](/definitions/20240819_definition_docker.md) containers. It provides the necessary libraries and tools to seamlessly access NVIDIA GPUs from within [Docker](/definitions/20240819_definition_docker.md) containers, allowing you to leverage GPU acceleration for your applications.
+
+![nvidia diagram](assets/20240820_gpu_utilization_img2.jpg)
+
+To enable GPU support in [Docker](/definitions/20240819_definition_docker.md) containers, you need to install the NVIDIA Container Toolkit:
+
+```bash
+# Configure the production repository
+curl -fsSL https://nvidia.github.io/libnvidia-container/gpgkey | sudo gpg --dearmor -o /usr/share/keyrings/nvidia-container-toolkit-keyring.gpg \
+  && curl -s -L https://nvidia.github.io/libnvidia-container/stable/deb/nvidia-container-toolkit.list | \
+    sed 's#deb https://#deb [signed-by=/usr/share/keyrings/nvidia-container-toolkit-keyring.gpg] https://#g' | \
+    sudo tee /etc/apt/sources.list.d/nvidia-container-toolkit.list
+
+# Install the NVIDIA Container Toolkit
+sudo apt-get update
+sudo apt-get install -y nvidia-container-toolkit
+
+# Configure the container runtime
+sudo nvidia-ctk runtime configure --runtime=docker
+
+# Restart Docker to apply changes
+sudo systemctl restart docker
+```
+
+### Verify GPU Configuration
+
+To verify that your [Docker](/definitions/20240819_definition_docker.md) setup can access the GPU, run the following command:
+
+```bash
+docker run --rm --gpus all nvidia/cuda:12.6.0-base-ubuntu24.04 nvidia-smi
+```
+
+If your setup is correct, the `nvidia-smi` command will display information about your GPU, confirming that the GPU is accessible inside [Docker](/definitions/20240819_definition_docker.md) containers.
+
+## Running LLM Inference in a Docker Container
+
+With your GPU-enabled [Docker](/definitions/20240819_definition_docker.md) environment ready, the next step is to develop a text generation script using an [LLM](/definitions/20240820_definition_large-language-model.md) and run it inside a [Docker](/definitions/20240819_definition_docker.md) container.
+
+### Create Daytona Workspace
+
+Before starting, make sure to create a Daytona workspace to set up a development environment:
+
+```bash
+daytona create https://github.com/giraffekey/gpu-container-example
+```
+
+This will create a controlled environment with the required Python dependencies installed.
+
+You can open the workspace in your preferred IDE with the following command:
+
+```bash
+daytona code
+```
+
+### Download the GPT-2 Model
+
+There should be this script in your opened workspace:
```

Comment:
or better, why don 't you guide us on how you did it, 


like create a github project:...

create a `.devcontainer` folder then a `devcontainer.json`
then create `download.py` file and add code to download GPT-2 model
run `python3 download.py`
then to test the model create a` inference.py` file and add code
run `python3 inference.py`
....

push code... `daytona create [repo_url]` etc
you get the gist.


I think your article is proper, but it lacks the handholding of replicating it; it's all over the place. 


like we add a `Dockerfile`. In the main project, you could add your Dockerfile in the .devcontainer folder. set bit up so that whenever we `daytona create [repo_url]`


once a user starts from github - daytona create, then they can start using 


also why are Monitoring GPU Utilization? talk about it
for Benchmarking and Performance Tuning, talk about it, the article is not cohesive, you have the whole building points, just organzie it.